### PR TITLE
Gas Station API: Auth middleware

### PR DESCRIPTION
**Pull request by grevych**

### CODE REVIEW

File: config/develop.js [hunk 1]
```diff
@@ -2,6 +2,9 @@ module.exports = {
   api: {
     port: 8080,
   },
+  entities: {
+    hostess: '5c4955c210728692df56a6a4',
```
Comment (dtoledo23) [inline config/develop.js]: What is this for?

File: api/gas/middlewares.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const jwks = require('jwks-rsa');
+
+const secret = jwks.expressJwtSecret({
+  cache: true,
+  rateLimit: true,
+  jwksRequestsPerMinute: 5,
+  jwksUri: 'https://sitmenow.auth0.com/.well-known/jwks.json',
+});
+
+const auth = jwt({
```
Comment (dtoledo23) [inline api/gas/middlewares.js]: Tsss 💯

File: api/gas/index.js [hunk 3]
```diff
@@ -33,23 +31,8 @@ module.exports = (stores, useCases) => {
...
-      hostessStore: stores.hostessStore,
-      branchStore: stores.branchStore,
-    });
-
-    useCase.execute()
-      .then(_ => res.json(_))
-      .catch(error => res.json(error))
-  });
-
+  /*
```
Comment (dtoledo23) [inline api/gas/index.js]: Delete commented code 🐷

File: api/gas/index.js [hunk 4]
```diff
@@ -117,14 +77,41 @@ module.exports = (stores, useCases) => {
...
+  app.put('/gasStations/:gasStationId', middlewares.auth, function(req, res) {
+    const useCase = new useCases.HostessToggleGasStation({
+      branchId: req.params.gasStationId,
+      hostessId: config.entities.hostess || req.user.sub,
+      hostessStore: stores.hostessStore,
+      branchStore: stores.branchStore,
+    });
+
+    useCase.execute()
+      .then(_ => res.json(_))
```
Comment (dtoledo23) [inline api/gas/index.js]: Why not `then(res.json)`?

File: api/gas/index.js [hunk 5]
```diff
@@ -117,14 +77,41 @@ module.exports = (stores, useCases) => {
...
   });
 
+  app.put('/gasStations/:gasStationId', middlewares.auth, function(req, res) {
+    const useCase = new useCases.HostessToggleGasStation({
+      branchId: req.params.gasStationId,
+      hostessId: config.entities.hostess || req.user.sub,
+      hostessStore: stores.hostessStore,
+      branchStore: stores.branchStore,
+    });
+
```
Comment (dtoledo23) [inline api/gas/index.js]: Shouldn't you do a modelToObject before sending right here? 

This returns 🤢 
```
{
    "_id": "5c49d5f785d72c75de5f4da5",
    "_name": "Branch Seed",
    "_address": "Branch Seed Address",
    "_coordinates": [
        10,
        10
    ],
    "_restaurant": {
        "_id": "5c49d5f785d72c75de5f4da4"
    },
    "waitingTurns": 0
}
```

File: api/gas/middlewares.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  secret,
+  audience: 'https://miturno.com.mx/api/v1/',
+  issuer: 'https://sitmenow.auth0.com/',
```
Comment (dtoledo23) [inline api/gas/middlewares.js]: 😍

File: config/develop.js (hunk 1 above)
Comment (grevych) [reply config/develop.js]: The way auth0 let us to use auth_tokens for testing doesn't include a user (at least not the free version) so if that's the case we force our middleware to use this ID which is the one used to create a branch in the seed process.

File: api/gas/index.js (hunk 5 above)
Comment (grevych) [reply api/gas/index.js]: That's the object. Take a look at the `Branch` implementation, you will see that most of the properties are private. Sorry for the missing `serializer` class. Maybe if you would have helped 2 weeks ago with this.. HDTPM

File: api/gas/index.js (hunk 3 above)
Comment (grevych) [reply api/gas/index.js]: Just need to figure out how to handle private and public logic in a same endpoint. Do you know how to do that?

Comment (dtoledo23) [reply api/gas/index.js]: Maybe implement a wrapper around your current auth and pass a parameter to tell if authentication should be enforced or not. But that is a problem for another time

**Pull request merged**